From 45443a4a66fb9436f87d906b00816cbf7ea2d586 Mon Sep 17 00:00:00 2001
From: thrishalig <thrishalig@ami.com>
Date: Thu, 6 Jul 2023 21:09:37 +0530
Subject: [PATCH] Fix for chassis identify ipmi standard commands giving
 unspecified error issue

issue URL: https://git.ami.com/megarac/onetree-team/onetree-core/core-manageability/-/issues/155

Tested:

root@intel-obmc:~# ipmitool chassis identify
Chassis identify interval: default (15 seconds)
root@intel-obmc:~# ipmitool chassis identify 0
Chassis identify interval: off
root@intel-obmc:~# ipmitool chassis identify 10
Chassis identify interval: 10 seconds
root@intel-obmc:~# ipmitool chassis identify force
Chassis identify interval: indefinite

Signed-off-by: thrishalig <thrishalig@ami.com>
---
 src/chassiscommands.cpp | 15 ++++++---------
 1 file changed, 6 insertions(+), 9 deletions(-)

diff --git a/src/chassiscommands.cpp b/src/chassiscommands.cpp
index 4996f43..be048c0 100644
--- a/src/chassiscommands.cpp
+++ b/src/chassiscommands.cpp
@@ -185,20 +185,17 @@ ipmi::RspType<> ipmiChassisIdentify(std::optional<uint8_t> interval,
                                     std::optional<uint8_t> force)
 {
     uint8_t identifyInterval = interval.value_or(defaultIdentifyTimeOut);
-    uint8_t forceIdentify = force.value();
+    uint8_t forceIdentify = force.value_or(0);
     
     enclosureIdentifyLed(ledIDOnObj, false);
     identifyTimer->stop();
 
-    if (identifyInterval || forceIdentify == 0x01)
+    if (identifyInterval || forceIdentify)
     {
-	if(forceIdentify != 0x00)
-	{
-	   if(forceIdentify != 0x01)
-	   {
-		return ipmi::responseInvalidFieldRequest();
-	   }
-	}
+    	if((forceIdentify >> 1) != 0)
+    	{
+	    	return ipmi::responseInvalidFieldRequest();
+     	}
 
         enclosureIdentifyLed(ledIDBlinkObj, true);
         if (forceIdentify)
-- 
2.25.1

